plugins/scanners/fossid/src/test/kotlin/FossIdTest.kt

```diff
@@ -990,50 +990,53 @@ class FossIdTest : WordSpec({
         }
 
         "delete newly triggered scans if a package cannot be scanned" {
+            val projectCode = PROJECT
```

Comment:
@nnobelis Please have a look at the changes in this test. It took me 2 hours to figure out why my change made this test fail, and to my understanding the reason is that it abused mocking to make the removed `createProjectCode` function behave in a different way than it would normally do to create a test scenario I don't fully understand. However, as all other tests in this class succeed I believe that my changes did not introduce any actual issues.

FYI, the PR is not complete, I will add one commit to make the `projectName` available as a built-in variable for the `namingScanPattern` (or do you think that is not required?), and another commit to refactor the URL mapping config to be compatible with the new plugin API.

Reply:
Ok, thanks, I will continue with the other changes then.

> You see that the project name is statically coming from the run parameters while #projectName refers to a buildin variable that does not exist 

That was just recently renamed to `repositoryName` so in the the version we use it still exists.